arcgis-ios-sdk-samples/Maps/Create and save a map/SaveAsViewController.swift

```diff
@@ -38,12 +47,46 @@ class SaveAsViewController: UITableViewController {
             presentAlert(message: "Please enter a title.")
             return
         }
-        
+        // Get the tags from the text field.
         let tags = tagsTextField.text?
             .components(separatedBy: ",")
             .map { $0.trimmingCharacters(in: .whitespacesAndNewlines) } ?? []
-        
+        // Get the item description from the text field.
         let itemDescription = descriptionTextField.text ?? ""
-        delegate?.saveAsViewController(self, didInitiateSaveWithTitle: title, tags: tags, itemDescription: itemDescription)
+        
+        delegate?.saveAsViewController(self, didInitiateSaveWithTitle: title, tags: tags, itemDescription: itemDescription, folder: selectedFolder)
+    }
+    
+    /// Present the list of folders to choose from.
+    func showFolderOptions() {
+        // Set index for the default option.
+        let selectedIndex = selectedFolderIndex ?? portalFolders.count
+        // The titles of the portal folders and a "None" option.
+        let folderTitles = portalFolders.map { $0.title! } + ["None"]
```

Comment:
```suggestion
        let folderTitles = portalFolders.map(\.title!) + ["None"]
```

Don't have to make this change. Just want to double check if we prefer the key path style.

Reply:
Does that work? If so, I didn't know keypaths to nullable properties could be force unwrapped like that.

Reply:
Seems to compile fine on my end. But I'm quite hesitant to write it like that

- I too have never seen key path unwrapping
- Even if the folder title is almost guaranteed not to be `nil`, I still don't like the ❗ . What about `let folderTitles = portalFolders.compactMap(\.title) + ["None"]`?

Reply:
Not a good idea to compact map because indices will be messed up in the event of `nil`. Better to just provide a default value.

```suggestion
        let folderTitles = portalFolders.map { $0.title ?? "" } + ["None"]
```